test/__init__.py

```diff
@@ -39,6 +39,20 @@
 LONG_TIMEOUT = 0.5 if os.environ.get("CI") else 0.01
 
 
+def _can_resolve(host):
+    """ Returns True if the system can resolve host to an address. """
+    try:
+        socket.getaddrinfo(host, None, socket.AF_UNSPEC)
+        return True
+    except socket.gaierror:
+        return False
+
+
+# Some systems might not resolve "localhost." correctly. We treat such
+# resolvers as unsupported. See https://github.com/urllib3/urllib3/issues/1809
+HAS_SUPPORTED_RESOLVER = _can_resolve("localhost.")
```

Comment:
I don't understand what a supported resolver is. Can we rename this to `RESOLVES_LOCALHOST_FQDN` or `RESOLVES_DOTTED_LOCALHOST`? And change the skipif reason to `"Can't resolve localhost."`.

Also, since 1/ we don't want to repeat ourselves and 2/ we want to be consistent with the skips below, we should write a function like `notBrotlipy` but adapted for this use case. Not sure about the name, maybe `resolvesLocalhostFQDN` or `resolvesDottedLocalhost`? This would allow us to just decorate the failing tests, like we do for all the other cases.

Reply:
Done, with `RESOLVES_LOCALHOST_FQDN` and `resolvesLocalhostFQDN`.